### ISSUE TITLE
Ray wrapper module to circumvent Ray easily

### ISSUE THREAD

**Issue by jammont** [isofit/isofit#301]
tl;dr: Create a wrapper module that can perform Ray-like operations so that isofit core development can continue using Ray syntax without having Ray enabled.

---

# The Problem

Isofit relies on Ray in several locations currently:
- `ray.get(...)`
  - https://github.com/isofit/isofit/blob/master/isofit/radiative_transfer/look_up_tables.py#L223
  - https://github.com/isofit/isofit/blob/master/isofit/radiative_transfer/sRTMnet.py#L179
  - https://github.com/isofit/isofit/blob/master/isofit/utils/atm_interpolation.py#L297
  - https://github.com/isofit/isofit/blob/master/isofit/utils/empirical_line.py.py#L422
  - https://github.com/isofit/isofit/blob/master/isofit/utils/extractions.py#L156
  - https://github.com/isofit/isofit/blob/master/isofit/utils/segment.py#L193
- `[@]ray.remote`
  - https://github.com/isofit/isofit/blob/master/isofit/core/isofit.py#L143
  - All of the `ray.get` scripts at various lines
- `ray.put(...)`
  - https://github.com/isofit/isofit/blob/master/isofit/core/isofit.py#L140
  - https://github.com/isofit/isofit/blob/master/isofit/utils/analytical_line.py#L139
  - https://github.com/isofit/isofit/blob/master/isofit/utils/extractions.py#L140

There is a debug flag in the configuration that can disable Ray for [isofit.py](https://github.com/isofit/isofit/blob/master/isofit/core/isofit.py#L139). This implementation only affects isofit.py, but the above scripts will instantiate a local Ray cluster despite the debug flag. 

# Proposed Solution

Rather than doing what isofit.py does using a if/else debug flag to determine how to run the code (with/out Ray), create a thin wrapper module with functions that act Ray-like. That is, isofit core code can continue using Ray syntax such as:
```python
id     = ray.put(obj)
func   = ray.remote(function)
jobs   = [func.remote(*args, id, **kwargs) for args, kwargs in iterable]
result = ray.get(jobs)
```
and the wrapper module would either submit these jobs to a Ray cluster if not in debug mode, otherwise run sequentially:
```python
class wrapped_Ray:
  def init(config):
    self.debug = config.debug

  def __getattr__(self, key):
    if self.debug:
      getattr(self, key)
    else:
      return getattr(ray, key)

  @staticmethod
  def put(obj):
    return obj

  @staticmethod
  def remote(func):
    func.remote = func
    return func

  @staticmethod
  def get(*args):
    return args
```
This won't work exactly but it gets the idea.

**Comment by pgbrodrick** [comment]
Solved with #333!
